### ISSUE TITLE
Use the cv splitter and scoring of skrub learners

### ISSUE THREAD

**Issue by jeromedockes** [probabl-ai/skore#2884]
it is possible in skrub DataOps to specify a cross-validation splitter and a scoring metric (and any additional arguments they need):

```python
import skore
import skrub
from sklearn.dummy import DummyClassifier
from sklearn.model_selection import LeaveOneGroupOut

df = skrub.datasets.toy_products()
data = skrub.var("df", df)
groups = data["seller"]
# Configure the cv splitter
X = data[["description", "price"]].skb.mark_as_X(
    cv=LeaveOneGroupOut(), split_kwargs={"groups": groups}
)
sample_weight = X["price"]
y = data["category"].skb.mark_as_y()
pred = (
    X.skb.apply(DummyClassifier(), y=y)
    # Configure scoring
    .skb.with_scoring("accuracy")
    .skb.with_scoring(
        "accuracy",
        kwargs={"sample_weight": sample_weight},
        name="weighted_accuracy",
    )
)

cv = pred.skb.cross_validate() # splits grouping by seller, scores with and without sample weights
print(cv)
```

```
   fit_time  score_time  test_accuracy  test_weighted_accuracy
0  0.001563    0.003420       0.666667                0.594595
1  0.001237    0.002599       0.666667                0.909091
```

At the moment skore ignores this configuration (it overrides it with its own defaults).

```python
report = skore.evaluate(pred.skb.make_learner(), data={"df": df})
summary = report.metrics.summarize().frame()
print(summary)
```

```
                             SkrubLearner
                                     mean
Metric           Label
Accuracy                         0.500000
Precision        electronics     0.500000
                 tools           0.000000
Recall           electronics     1.000000
                 tools           0.000000
ROC AUC                          0.500000
Log loss                         0.836988
Brier score                      0.312500
Fit time (s)                     0.002137
Predict time (s)                 0.000943
```

It would be helpful if it could use the configured splitter by default when it exists, and either use the configured scorers or add them to the default ones in .metrics.summarize() (e.g. weighted_accuracy would show up in the summary above, and splitting would use the groups)

**Comment by jeromedockes** [comment]
more generally we should probably also include the estimator's `score()` in the displayed metrics

**Comment by jeromedockes** [comment]
about the fact that currently skore interprets dicts as `{label: score}` and scikit-learn as `{metric_name: score}`: scikit-learn functions like precision that need to return 1 value / label return an array. so we could keep arrays for labels and dicts for metric names; ie the output of a multimetric scorer for a multiclass problem could look like `{'accuracy': 0.3, 'precision': [0.3, 0.3, 0.3]}` . it could also be `[0.3, 0.3, 0.3]` (single metric, one entry / label) or `{'accuracy': 0.3, 'log_loss': 0.2}` (multiple metrics, one value for all labels) or `0.2` (single metric, only one value). internally skore can then consolidate this into something more explicit which also includes the actual labels.

an alternative could be forcing a custom skore type for label: score mappings. e.g. `{'accuracy': 0.2}` -> metric name, `skore.PerLabelScores({'yes': .3, 'no': .3, 'maybe': .3})` -> label. this is a bit cumbersome but possible because this convention is only used in skore, so most likely only skore internal code would need to use that type

**Comment by glemaitre** [comment]
I was looking more at the problematic of the `splitter`. Basically, right now the only way that we could support `groups` would be to pass them and use the `mark_as_X`.

However, we are storing the indices of the cross-validation and I think that it is currently a limitation. Also we should be defining the expected behaviour: should we pass to both `mark_as_X` and `evaluate`. If at both, which one take precedence. If only one is provided, how to not get confused with `None` which means a single estimator.
